### PR TITLE
upgrade to 3-0-stable

### PR DESCRIPTION
**Pull request by nimish13**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ nbproject
 pkg
 *.swp
 spec/dummy
+.byebug_history
```
Comment (siddharth28) [inline .gitignore]: @nimish13 is this required?

File: Gemfile [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
 group :test do
-  gem 'rspec-rails', '~> 2.12.0'
-  gem 'factory_girl_rails', '~> 4.2.1'
-  gem 'email_spec', '~> 1.2.1'
-
+  gem 'test-unit'
+  gem 'rspec-rails', '~> 3.3.3'
+  gem 'factory_girl_rails', '~> 4.5.0'
+  gem 'email_spec'
+  gem 'byebug'
```
Comment (siddharth28) [inline Gemfile]: @nimish13 is `byebug` required?

File: Gemfile [hunk 3]
```diff
@@ -1,24 +1,26 @@
 source 'https://rubygems.org'
 
 group :test do
-  gem 'rspec-rails', '~> 2.12.0'
-  gem 'factory_girl_rails', '~> 4.2.1'
-  gem 'email_spec', '~> 1.2.1'
-
+  gem 'test-unit'
```
Comment (siddharth28) [inline Gemfile]: @nimish13 why do we need `test-unit` ?

File: app/assets/javascripts/spree/backend/spree_paga.js [hunk 4]
```diff
@@ -0,0 +1 @@
+s
```
Comment (siddharth28) [inline app/assets/javascripts/spree/backend/spree_paga.js]: @nimish13 why have we added this?

File: app/models/spree/order_decorator.rb [hunk 5]
```diff
@@ -27,7 +27,7 @@ def payment_or_complete_or_pending?
 
   def finalize_order
     paga_payment.complete!
-    update_attributes({:state => "complete", :completed_at => Time.now}, :without_protection => true)
+    update_attributes(state: "complete", completed_at: Time.now)
```
Comment (siddharth28) [inline app/models/spree/order_decorator.rb]: @nimish13 i think we should use `Time.current` here.

File: app/models/spree/paga_notification.rb [hunk 6]
```diff
@@ -19,7 +19,7 @@ def self.save_with_params(notification_params)
   private
 
   def update_transaction_status
-    paga_transaction = Spree::PagaTransaction.where(:transaction_id => self.transaction_id).first
+    paga_transaction = Spree::PagaTransaction.where(transaction_id: self.transaction_id).first
```
Comment (siddharth28) [inline app/models/spree/paga_notification.rb]: @nimish13 i think we can use `find_by` here.

File: app/models/spree/paga_transaction.rb [hunk 7]
```diff
@@ -78,7 +78,7 @@ def update_payment_source
   end
 
   def update_transaction_status
-    paga_notification = Spree::PagaNotification.where(:transaction_id => self.transaction_id).first
+    paga_notification = Spree::PagaNotification.where(transaction_id: self.transaction_id).first
```
Comment (siddharth28) [inline app/models/spree/paga_transaction.rb]: @nimish13 DRY not followed `update_transaction_status`.
Similar method in `Spree::PagaNotification`
Also use `find_by` here.

Comment (siddharth28) [inline app/models/spree/paga_transaction.rb]: @nimish13 i think we can use  `find_by` here.

File: app/controllers/spree/checkout_controller_decorator.rb [hunk 8]
```diff
@@ -39,9 +39,9 @@ def paga_notification
 
     def redirect_to_paga_if_payment_by_paga
       if payment_page? && payment_attributes = paga_payment_attributes(params[:order][:payments_attributes])
-        payment_method = Spree::PaymentMethod.where(:id => payment_attributes[:payment_method_id]).first
+        payment_method = Spree::PaymentMethod.where(id: payment_attributes[:payment_method_id]).first
```
Comment (siddharth28) [inline app/controllers/spree/checkout_controller_decorator.rb]: @nimish13 i think we can use `find_by` here.

File: spec/controllers/spree/checkout_controller_decorator_spec.rb [hunk 9]
```diff
@@ -48,32 +51,32 @@ def send_request(params = {})
...
         send_request(:order => { :payments_attributes => [{:payment_method_id => paga_payment_method.id}]}, :state => "payment")
       end
 
       context 'when payment_method kind_of Spree::PaymentMethod::Paga' do
-        it "should_receive update_attributes" do
-          order.should_receive(:update_attributes).and_return(true)
-          controller.should_receive(:object_params)
+        it "should_receive update_from_params" do
+          expect(order).to receive(:update_from_params).and_return(true)
+          # expect(controller).to receive(:object_params)
```
Comment (siddharth28) [inline spec/controllers/spree/checkout_controller_decorator_spec.rb]: @nimish13 why is commented code present?

File: spec/spec_helper.rb [hunk 10]
```diff
@@ -67,4 +67,15 @@
   end
 
   config.fail_fast = ENV['FAIL_FAST'] || false
+
```
Comment (siddharth28) [inline spec/spec_helper.rb]: @nimish13  why is commented code present?

**Comment by siddharth28** [comment]
@nimish13 fix the warnings in the spec.

File: app/models/spree/order_decorator.rb [hunk 11]
```diff
@@ -3,15 +3,15 @@
   has_many :paga_transactions
 
   def paga_payment
-    @paga_payment_method = Spree::PaymentMethod::Paga.where(:environment => Rails.env).first
+    @paga_payment_method = Spree::PaymentMethod::Paga.first
     payments.where("payment_method_id = #{@paga_payment_method.id} and state in ('checkout', 'pending', 'processing')").first if @paga_payment_method
   end
 
-  scope :not_pending, where('state != ?', "pending")
+  scope :not_pending, -> { where('state != ?', "pending") }
```
Comment (priyank-gupta) [inline app/models/spree/order_decorator.rb]: @nimish13 We can use `where.not`.

File: spec/controllers/spree/admin/paga_transaction_controller_spec.rb [hunk 12]
```diff
@@ -5,28 +5,24 @@
 
   describe 'index' do
     def send_request
-      get :index, :page => "1", :use_route => "spree"
+      get :index, :page => "1"
```
Comment (priyank-gupta) [inline spec/controllers/spree/admin/paga_transaction_controller_spec.rb]: @nimish13 Use new hash syntax

**Pull request merged**